In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import StandardScaler, MinMaxScaler, RobustScaler
from sklearn.metrics import accuracy_score, f1_score, recall_score, precision_score, confusion_matrix
import seaborn as sns
import matplotlib.pyplot as plt
from pycaret.classification import *
from sklearn.model_selection import StratifiedKFold
from datetime import datetime
df = pd.read_csv('C:/Users/User/Downloads/Social_Network_Ads.csv')

df['Gender'] = df['Gender'].map({'Male': 0, 'Female': 1})

In [3]:
df

Gender  Age  EstimatedSalary  Purchased
0         0   19            19000          0
1         0   35            20000          0
2         1   26            43000          0
3         1   27            57000          0
4         0   19            76000          0
..      ...  ...              ...        ...
395       1   46            41000          1
396       0   51            23000          1
397       1   50            20000          1
398       0   36            33000          0
399       1   49            36000          1

[400 rows x 4 columns]

In [2]:
df = df.drop(['User ID'], axis=1)

In [4]:
import statsmodels.api as sm 

In [5]:
Xtrain = df.drop(['Purchased'], axis=1)
ytrain = df[['Purchased']] 
   
log_reg = sm.Logit(ytrain, Xtrain).fit() 

Optimization terminated successfully.
         Current function value: 0.677546
         Iterations 4


In [6]:
print(log_reg.summary()) 

                           Logit Regression Results                           
Dep. Variable:              Purchased   No. Observations:                  400
Model:                          Logit   Df Residuals:                      397
Method:                           MLE   Df Model:                            2
Date:                Wed, 05 Feb 2025   Pseudo R-squ.:                -0.03923
Time:                        17:22:44   Log-Likelihood:                -271.02
converged:                       True   LL-Null:                       -260.79
Covariance Type:            nonrobust   LLR p-value:                     1.000
                      coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------
Gender             -0.5371      0.201     -2.666      0.008      -0.932      -0.142
Age                -0.0002      0.006     -0.035      0.972      -0.012       0.011
EstimatedSalary  6.209e-07   2.8

## Logit Regression Results 해석
- Pseudo R-squared: -0.03923
→ 현재 모델의 적합도와 모델의 설명력이 낮거나 개선이 필요함.

- Age: p-value: 0.972, z 값: 약 0
→ Age는 구매 여부에 통계적으로 유의한 영향을 미치지 않습니다.

- EstimatedSalary: p-value: 0.826, z 값: 약 0
→ EstimatedSalary 역시 구매 여부에 유의한 영향을 주지 않는 것으로 나타납니다.

- Gender: 계수: -0.5371, p-value: 0.008, 변수 매핑: {'Male': 0, 'Female': 1}

- 오즈비 계산:
exp
⁡
(
−
0.5371
)
≈
0.5844
exp(−0.5371)≈0.5844

- 해석:
남성을 기준(reference)으로 했을 때, 여성이 구매할 오즈가 약 0.5844배입니다.
이는 여성이 구매할 가능성이 남성보다 약 41.6% 낮음을 의미합니다
(즉, 
1
−
0.5844
≈
0.4156
1−0.5844≈0.4156).